spinalcordtoolbox/vertebrae/detect_c2c3.py

```diff
@@ -139,8 +138,21 @@ def detect_c2c3_from_file(fname_im, fname_seg, contrast, fname_c2c3=None, verbos
     nii_im = Image(fname_im)
     nii_seg = Image(fname_seg)
 
+    remove_temp_files = int(remove_temp_files)
+
     # detect C2-C3
-    nii_c2c3 = detect_c2c3(nii_im, nii_seg, contrast, verbose=verbose)
+    if remove_temp_files:
+        nii_c2c3 = detect_c2c3(nii_im.copy(), nii_seg, contrast, verbose=verbose)
+    else:
+        nii_c2c3, nii_midSlice, nii_mask, nii_after_postPro, nii_before_postPro = detect_c2c3(nii_im.copy(),
+                                                                                                nii_seg,
+                                                                                                contrast,
+                                                                                                remove_temp_files=remove_temp_files,
+                                                                                                verbose=verbose)
+        nii_midSlice.save(os.path.join(os.path.dirname(nii_im.absolutepath), "tmp_midSlice.nii.gz"))
```

Comment:
now that's a bit confusing. Here we explicitly save debugging file by setting "remove_temp_files" to 0, which is counter-intuitive. 

the variable `remove_temp_files` should only be used for the action "remove temp file".

if we want to save debugging files, we should use another variable (e.g. debugging), or, as mostly done in SCT, use verbose=2 mode.